src/test/java/bio/terra/workspace/service/datareference/model/ReferenceObjectTest.java

```diff
@@ -31,4 +31,52 @@ public void SnapshotReferenceSerializationBackwardsCompatible() {
     assertEquals("foo", snapshotReference.instanceName());
     assertEquals("bar", snapshotReference.snapshot());
   }
+
+  /**
+   * This is a simple test for verifying that GoogleBucketReference's toJson() and fromJson() are
+   * compatible. If this failing, you've likely broken one or both of those methods.
+   */
+  @Test
+  public void GoogleBucketReferenceSerializationWorks() {
+    GoogleBucketReference bucket = GoogleBucketReference.create(/* bucketName= */ "foo");
+    assertEquals(bucket, ReferenceObject.fromJson(bucket.toJson()));
+  }
+
+  /**
+   * Hard code serialized values to check that code changes do not break backwards compatibility of
+   * stored JSON values. If this test fails, your change may not work with existing databases.
+   */
+  @Test
+  public void GoogleBucketReferenceSerializationBackwardsCompatible() {
+    GoogleBucketReference bucketReference =
+        (GoogleBucketReference)
+            ReferenceObject.fromJson(
+                "{\"@type\":\"GoogleBucketReference\",\"bucketName\":\"foo\"}");
+    assertEquals("foo", bucketReference.bucketName());
+  }
+
+  /**
+   * This is a simple test for verifying that BigQueryDataset's toJson() and fromJson() are
+   * compatible. If this failing, you've likely broken one or both of those methods.
+   */
+  @Test
+  public void BigQueryDatasetReferenceSerializationWorks() {
+    BigQueryDatasetReference dataset =
+        BigQueryDatasetReference.create(/* projectId= */ "foo", /* datasetName= */ "bar");
+    assertEquals(dataset, ReferenceObject.fromJson(dataset.toJson()));
+  }
+
+  /**
+   * Hard code serialized values to check that code changes do not break backwards compatibility of
+   * stored JSON values. If this test fails, your change may not work with existing databases.
+   */
+  @Test
+  public void BigQueryDatasetReferenceSerializationBackwardsCompatible() {
+    BigQueryDatasetReference datasetReference =
+        (BigQueryDatasetReference)
+            ReferenceObject.fromJson(
+                "{\"@type\":\"BigQueryDatasetReference\",\"projectId\":\"foo\",\"datasetName\":\"bar\"}");
```

Comment:
any more complicated than this and we should just have JSON files in the test references

Reply:
I think these are simple enough JSON structures to leave as-is, and I think they provide a quick smoke test level of "do my changes break serialization"

Reply:
You must really like backslashes.